lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -163,16 +160,16 @@ def base_url
       end
 
       def avs_result(action, succeeded, response)
-        if succeeded
-          action == :purchase ? AVSResult.new(code: response["card"]["avsCheck"]) : nil
+        if succeeded && action == :purchase || :authorize
+          AVSResult.new(code: response['card']['avsCheck']) if response['card'] && response['card']['avsCheck']
```

Comment:
This is personal preference, but this condition is getting sufficiently complex I'd generally prefer it not to be a guard clause.

Reply:
Agreed -- removed.